test/support/mocks/graph-client.service.mock.ts

```diff
@@ -0,0 +1,57 @@
+import { Client, GraphRequest } from '@microsoft/microsoft-graph-client';
+import { when } from 'jest-when';
+
+class MockClient {
+  api: jest.Mock;
+  constructor() {
+    this.api = jest.fn();
+  }
+}
+
+class MockRequest {
+  get: jest.Mock<any, any, any>;
+  filter: jest.Mock<any, any, any>;
+  expand: jest.Mock<any, any, any>;
+  constructor() {
+    this.get = jest.fn();
+
+    this.filter = jest.fn();
+    this.expand = jest.fn();
+  }
+}
+
+export class MockGraphClientService {
+  client: Client;
+  request: GraphRequest;
+  constructor() {
+    this.client = new MockClient() as unknown as Client;
+    this.request = new MockRequest() as unknown as GraphRequest;
+  }
+
+  getApiRequestObject() {
+    return this.request;
+  }
+
+  mockSuccessfulGraphApiCallWithPath(path: string) {
+    when(this.client.api).calledWith(path).mockReturnValueOnce(this.request);
+    return this;
+  }
```

Comment:
Not an issue for this PR, but to look out for on future tickets: since `client.api` is always returning the same `request` object, we're going to get conflicting mocking/assertions when we use this for endpoints that make more than one call to Microsoft Graph. We'll need to change this as soon as we have endpoints like that